iree/compiler/Codegen/SPIRV/SPIRVVectorizeLoadStore.cpp

```diff
@@ -451,8 +451,8 @@ struct ScalarizeVectorTransferWrite final
   LogicalResult matchAndRewrite(vector::TransferWriteOp writeOp,
                                 PatternRewriter &rewriter) const override {
     VectorType vectorType = writeOp.getVectorType();
-    if (vectorType.getRank() != 1 ||
-        !writeOp.getPermutationMap().isMinorIdentity())
+    auto map = writeOp.getPermutationMap();
+    if (vectorType.getRank() != 1 || !map.isProjectedPermutation())
```

Comment:
not that supporting the general case wouldn't be super hard. It would just require handling `0` which means don't touch the index.

Reply:
Oh you mean 0-D vector? Haven't seen instances of it in real yet.